src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java

```diff
@@ -73,6 +74,8 @@
 
     private final List<PodImagePullSecret> imagePullSecrets = new ArrayList<PodImagePullSecret>();
 
+    private final List<ToolLocationNodeProperty> nodeProperties = Collections.emptyList();
```

Comment:
I don't think this will work in an upgrade, nodeProperties will be null. 
Either implement readResolve to set it to emptyList if null https://github.com/jenkinsci/kubernetes-plugin/blob/master/src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java#L348
or check in getNodeProperties